clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -104,6 +104,7 @@ static ResourceClass getResourceClass(RegisterType RT) {
   case RegisterType::I:
     llvm_unreachable("unexpected RegisterType value");
   }
+  llvm_unreachable("unhandled RegisterType");
```

Comment:
Whats the difference between an unexpected and an unhandled register type? Why couldn't we have just done a default case?